tasks/upload.yml

```diff
@@ -22,7 +22,7 @@
     interface: "{{ os_images_interface | default(omit, true) }}"
     name: "{{ item.name ~ '-kernel' }}"
     state: present
-    is_public: "{{ item.is_public | default(os_images_public) | bool }}"
+    is_public: "{{ (item.is_public or item.visibility == 'public') | default(os_images_visibility == 'public') | bool }}"
```

Comment:
think this is what we need, should be backwards compatible as `os_images_visibility` is set by the value provided in `os_images_public`